iOSClient/Trash/Cell/NCTrashGridCell.swift

```diff
@@ -0,0 +1,287 @@
+//
+//  NCTrashGridCell.swift
+//  Nextcloud
+//
+//  Created by Marino Faggiana on 19/03/2024.
+//  Copyright © 2024 Marino Faggiana. All rights reserved.
+//
+//  Author Marino Faggiana <marino.faggiana@nextcloud.com>
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import UIKit
+
+class NCTrashGridCell: UICollectionViewCell, UIGestureRecognizerDelegate, NCCellProtocol, NCTrashCellProtocol {
+
+    @IBOutlet weak var imageItem: UIImageView!
+    @IBOutlet weak var imageSelect: UIImageView!
+    @IBOutlet weak var imageStatus: UIImageView!
+    @IBOutlet weak var imageFavorite: UIImageView!
+    @IBOutlet weak var imageLocal: UIImageView!
+    @IBOutlet weak var labelTitle: UILabel!
+    @IBOutlet weak var labelInfo: UILabel!
+    @IBOutlet weak var labelSubinfo: UILabel!
+    @IBOutlet weak var buttonMore: UIButton!
+    @IBOutlet weak var imageVisualEffect: UIVisualEffectView!
+
+    var objectId = ""
+    var indexPath = IndexPath()
+    private var user = ""
+
+    weak var gridCellDelegate: NCTrashGridCellDelegate?
+    var namedButtonMore = ""
+
+    var fileObjectId: String? {
+        get { return objectId }
+        set { objectId = newValue ?? "" }
+    }
+    var filePreviewImageView: UIImageView? {
+        get { return imageItem }
+        set { imageItem = newValue }
+    }
+    var fileUser: String? {
+        get { return user }
+        set { user = newValue ?? "" }
+    }
+    var fileTitleLabel: UILabel? {
+        get { return labelTitle }
+        set { labelTitle = newValue }
+    }
+    var fileInfoLabel: UILabel? {
+        get { return labelInfo }
+        set { labelInfo = newValue }
+    }
+    var fileSubinfoLabel: UILabel? {
+        get { return labelSubinfo }
+        set { labelSubinfo = newValue }
+    }
+    var fileSelectImage: UIImageView? {
+        get { return imageSelect }
+        set { imageSelect = newValue }
+    }
+    var fileStatusImage: UIImageView? {
+        get { return imageStatus }
+        set { imageStatus = newValue }
+    }
+    var fileLocalImage: UIImageView? {
+        get { return imageLocal }
+        set { imageLocal = newValue }
+    }
+    var fileFavoriteImage: UIImageView? {
+        get { return imageFavorite }
+        set { imageFavorite = newValue }
+    }
```

Comment:
This is a lot of boilerplate that is useless IMO, it's hard to remove as it's in many parts of the code, but good to remove it nontheless